app/src/ui/history/commit-list.tsx

```diff
@@ -171,6 +179,19 @@ export class CommitList extends React.Component<ICommitListProps, {}> {
     )
   }
 
+  private onSquash = (toSquash: ReadonlyArray<Commit>, squashOnto: Commit) => {
+    const indexes = [...toSquash, squashOnto].map(v =>
+      this.props.commitSHAs.findIndex(sha => sha === v.sha)
+    )
+    const maxIndex = Math.max(...indexes)
+    const lastIndex = this.props.commitSHAs.length - 1
+    /* If the commit is the first commit in the branch, you cannot reference it
+    using the sha, you must use the --root flag */
+    const lastRetainedCommitRef =
+      maxIndex === lastIndex ? '--root' : `${this.props.commitSHAs[maxIndex]}^`
```

Comment:
What is `--root` exactly? Is it something git specific? If so, is it ok to "leak" that at this point or is there a better way to indicate this (like a `null`)? 🤔

Reply:
Here is some documentation on the `--root flag` for reference. https://git-scm.com/docs/git-rebase/2.19.0#Documentation/git-rebase.txt---root

Refactored to `null` - I thought of doing that but thought specifying `--root` was more direct/readable.. But, can see where we don't want to pass `git` specific flags around and `null` does make since because the lastRetainedCommitRef is the commit before the last commit in squash operation which does not exist in this context so `null` make sense.